docs/zephyr_02_functional_requirements.sdoc

```diff
@@ -1723,19 +1762,45 @@ Zephyr shall provide an interface to schedule multiple threads.
 DISCUSSION_DATE: >>>
 pf-ok
 <<<
+RELATIONS:
+- TYPE: Parent
+  VALUE: ZEP-5
 
 [REQUIREMENT]
-UID: ZEP-121
+UID: ZEP-125
 STATUS: Draft
 TYPE: Functional
 COMPONENT: Threads
-TITLE: Threads - TBD
+TITLE: Thread Options
 STATEMENT: >>>
-??? Can Zephyr change priviledge level of a thread once created??
+The Zephyr kernel shall support a set of thread options.
 <<<
-DISCUSSION_DATE: >>>
-TBD
+USER_STORY: >>>
+As a Zephyr OS user, I want to be able to pass specific option to a thread.
 <<<
+RELATIONS:
+- TYPE: Parent
+  VALUE: ZEP-123
+- TYPE: Parent
+  VALUE: ZEP-5
+
+[REQUIREMENT]
+UID: ZEP-126
+STATUS: Draft
+TYPE: Functional
+COMPONENT: Threads
+TITLE: Thread Custom Data
```

Comment:
**question:** Is that the same as Thread Local Storage or something different?

Reply:
I think it is a little bit different. It needs configuration and is only accessible by the thread itself and it is not available in ISRs